torch/utils/data/dataset.py

```diff
@@ -235,6 +236,55 @@ def set_reduce_ex_hook(cls, hook_fn):
             raise Exception("Attempt to override existing reduce_ex_hook")
         IterableDataset.reduce_ex_hook = hook_fn
 
+    @property
+    def slice(self):
+        return _DataPipeSlice(self)
+
+
+class SliceIterDataPipe(IterableDataset):
```

Comment:
While reviewing a PR from torchvision, there is a different definition for `slice`. https://github.com/pytorch/vision/pull/4512

I think the following patten may be more suitable for `IterDataPipe`:
- `dp[0, 1]` -> select columns
- `dp.slice(0, 1000, 2)` -> yield item from start (0) to end (1000) in step (2)